plugin/trino-mongodb/src/test/java/io/trino/plugin/mongodb/TestMongoProjectionPushdownPlans.java

```diff
@@ -0,0 +1,295 @@
+                    ImmutableMap.of("mongodb.connection-url", server.getConnectionString().toString()));
+            // Put an dummy schema collection because MongoDB doesn't support a database without collections
+            client.getDatabase(SCHEMA).createCollection("dummy");
+        }
+        catch (Throwable e) {
+            closeAllSuppress(e, queryRunner);
+            throw e;
+        }
+        return queryRunner;
+    }
+
+    @AfterClass(alwaysRun = true)
+    public final void destroy()
+    {
+        server.close();
+        server = null;
+        client.close();
+        client = null;
+    }
+
+    @Test
+    public void testPushdownDisabled()
+    {
+        String tableName = "test_pushdown_disabled_" + randomNameSuffix();
+
+        Session session = Session.builder(getQueryRunner().getDefaultSession())
+                .setCatalogSessionProperty(CATALOG, "projection_pushdown_enabled", "false")
+                .build();
+
+        getQueryRunner().execute("CREATE TABLE " + tableName + " (col0) AS SELECT CAST(row(5, 6) AS row(a bigint, b bigint)) AS col0 WHERE false");
+
+        assertPlan(
+                "SELECT col0.a expr_a, col0.b expr_b FROM " + tableName,
+                session,
+                any(
+                        project(
+                                ImmutableMap.of("expr", expression("col0[1]"), "expr_2", expression("col0[2]")),
+                                tableScan(tableName, ImmutableMap.of("col0", "col0")))));
+    }
+
+    @Test
+    public void testDereferencePushdown()
+    {
+        String tableName = "test_simple_projection_pushdown" + randomNameSuffix();
+        QualifiedObjectName completeTableName = new QualifiedObjectName(CATALOG, SCHEMA, tableName);
+
+        getQueryRunner().execute("CREATE TABLE " + tableName + " (col0, col1)" +
+                " AS SELECT CAST(row(5, 6) AS row(x BIGINT, y BIGINT)) AS col0, BIGINT '5' AS col1");
+
+        Session session = getQueryRunner().getDefaultSession();
+
+        Optional<TableHandle> tableHandle = getTableHandle(session, completeTableName);
+        assertThat(tableHandle).as("expected the table handle to be present").isPresent();
+
+        Map<String, ColumnHandle> columns = getColumnHandles(session, completeTableName);
+
+        MongoColumnHandle column0Handle = (MongoColumnHandle) columns.get("col0");
+        MongoColumnHandle column1Handle = (MongoColumnHandle) columns.get("col1");
+
+        MongoColumnHandle columnX = createProjectedColumnHandle(column0Handle, ImmutableList.of("x"), BIGINT);
+        MongoColumnHandle columnY = createProjectedColumnHandle(column0Handle, ImmutableList.of("y"), BIGINT);
+
+        // Simple Projection pushdown
+        assertPlan(
+                "SELECT col0.x expr_x, col0.y expr_y FROM " + tableName,
+                any(tableScan(
+                        equalTo(((MongoTableHandle) tableHandle.get().getConnectorHandle()).withProjectedColumns(Set.of(columnX, columnY))),
```

Comment:
Can we extract `tableHandle.get().getConnectorHandle()` as a seperate variable and use them in the tests.